validator/keymanager/remote-web3signer/keymanager.go

```diff
@@ -38,54 +53,323 @@ type SetupConfig struct {
+	if err != nil {
+		return errors.Wrap(err, "could not stat remote signer public key file")
 	}
+	initialFileSize := initialFileInfo.Size()
+	if err := watcher.Add(km.keyFilePath); err != nil {
+		return errors.Wrap(err, "could not add file to file watcher")
+	}
+	log.WithField("path", km.keyFilePath).Info("Successfully initialized file watcher")
+	km.retriesRemaining = maxRetries // reset retries to default
+	// reinitialize keys if watcher reinitialized
+	if km.areEmptyPublicKeys() {
+		_, fk, err := km.readKeyFile()
+		if err != nil {
+			return errors.Wrap(err, "could not read key file")
+		}
+		maps.Copy(fk, km.flagLoadedKeysMap)
+		if err = km.savePublicKeysToFile(fk); err != nil {
+			return errors.Wrap(err, "could not save public keys to file")
+		}
+		km.updatePublicKeys(maps.Values(fk))
+	}
+	for {
+		select {
+		case e, ok := <-watcher.Events:
+			if !ok { // Channel was closed (i.e. Watcher.Close() was called).
+				log.Info("Closing file watcher")
+				return nil
+			}
+			log.WithFields(logrus.Fields{
+				"event": e.Name,
+				"op":    e.Op.String(),
+			}).Debug("Remote signer key file event triggered")
+			if e.Has(fsnotify.Remove) {
+				return errors.New("remote signer key file was removed")
+			}
+			currentFileInfo, err := os.Stat(km.keyFilePath)
+			if err != nil {
+				return errors.Wrap(err, "could not stat remote signer public key file")
+			}
+			if currentFileInfo.Size() != initialFileSize {
+				log.Info("Remote signer key file updated")
+				fileKeys, _, err := km.readKeyFile()
+				if err != nil {
+					return errors.New("could not read key file")
+				}
+				// prioritize file keys over flag keys
+				if len(fileKeys) == 0 {
+					log.Warnln("Remote signer key file no longer has keys, defaulting to flag provided keys")
+					fileKeys = maps.Values(km.flagLoadedKeysMap)
+				}
+				currentKeys, err := km.FetchValidatingPublicKeys(ctx)
+				if err != nil {
+					return errors.Wrap(err, "could not fetch current keys")
+				}
+				if !slices.Equal(currentKeys, fileKeys) {
+					km.updatePublicKeys(fileKeys)
+				}
+				initialFileSize = currentFileInfo.Size()
+			}
+		case err, ok := <-watcher.Errors:
+			if !ok { // Channel was closed (i.e. Watcher.Close() was called).
+				log.Info("Closing file watcher")
+				return nil
+			}
+			return errors.Wrap(err, "could not watch for file changes")
+		case <-ctx.Done():
+			return nil
+		}
+	}
+}
+
+func (km *Keymanager) updatePublicKeys(keys [][48]byte) {
+	km.lock.Lock()
+	defer km.lock.Unlock()
+	km.providedPublicKeys = keys
+	km.accountsChangedFeed.Send(keys)
+	log.WithField("count", len(km.providedPublicKeys)).Debug("Updated public keys")
+}
+
+// FetchValidatingPublicKeys fetches the validating public keys
+func (km *Keymanager) FetchValidatingPublicKeys(_ context.Context) ([][fieldparams.BLSPubkeyLength]byte, error) {
+	km.lock.RLock()
+	defer km.lock.RUnlock()
+	log.Debug("Fetched validating public keys", "count", len(km.providedPublicKeys))
```

Comment:
```suggestion
	log..WithField("count", len(km.providedPublicKeys)).Debug("Fetched validating public keys")
```
Was this supposed to be like that?

Reply:
with field works too